### DISCUSSION TITLE
Import massive amounts of recipes

### DISCUSSION THREAD

**Discussion by Ultragian** [TandoorRecipes/recipes#2803] [Q&A]
Is there a way to import a massive amount of recipes at once?

I have multiple formats, MMF isn't working.
I was thinking about pasting and converting everything to JSON with ChatGPT but Tandoor only detects on recipe at a time.
It would be complete madness to add my library of recipes one by one.
By massive I mean literally millions of recipes.
ChatGPT already limits me when asking for recipe conversions, doing them individually is insane.

I barely added a tenth of a single one of my cocktail recipe files.

**Comment by smilerz** [comment]
I'm not sure what you mean by MMF?  But if you are saying that your recipes are not in a standard format, then no, there isn't really a path towards automating importing them to Tandoor.

If you can standardize the format you can either:
a) Put it in one of the app formats that Tandoor can import (list app tab at `/data/import/url`)
b) Write a script that leverages the recipe API endpoint

**Reply by Ultragian** [reply]
Hmmm, I'm not experienced enough to write code yet.
MMF is the MealMaster file format. Nevermind, I managed to import it into Paprika and importing into Tandoor now. Looking for Accuchef file format support now.

**Reply by Ultragian** [reply]
Basically I have hundreds of recipe files, each with hundreds/thousands of recipes. But the files are in multiple formats. Not all of them are compatible. MMF is one of them. Tandoor can't seem to import multiple recipes in the MMF format, I'd have to do them one by one. But it works in Paprika format. And Paprika can import whole libraries in MMF. It just can't do Accuchef.
